### ISSUE TITLE
Inconsistent checkbox focus ring behaviour

### ISSUE THREAD

**Issue by Shankrie** [chakra-ui/chakra-ui#8000]
### Description

As you can see in the provided sandbox, the focus ring on checkboxes works differently when the checkbox is a descendant of an element with `tabindex` attribute.

Normally the checkbox does not have a focus ring when it is clicked with a mouse and the focus ring appears only when the checkbox is focused with keyboard. I believe this is the expected behaviour of an element with focus-visible styling. However, if the checkbox component is placed inside the element with `tabindex` attribute the focus ring appears on the first mouse click, then disappears if clicked again.

This problem is immediately apparent in modals because Chakra's modal contains elements with `tabindex` attribute.

NOTE: `tabindex` can be multiple levels above the checkbox, it does not have to be a direct parent

NOTE: the value of `tabindex` does not matter, the issue is reproducible with `-1`, `0` and other values

### Link to Reproduction

https://codesandbox.io/s/async-darkness-gwd7tg?file=/App.tsx

### Steps to reproduce

1. Go to the sandbox example: https://codesandbox.io/s/async-darkness-gwd7tg?file=/App.tsx
2. Click on first checkbox - it works as expected (no focus ring).
3. Click on second checkbox. The focus ring is visible.

### Chakra UI Version

2.8.1

### Browser

Google Chrome 116

### Operating System

- [X] macOS
- [ ] Windows
- [ ] Linux

### Additional Information

Recording: https://github.com/chakra-ui/chakra-ui/assets/807655/9c99dba4-5c22-4342-9b5f-cddabe74091c
